src/python/WMCore/WMSpec/Steps/Executors/LogArchive.py

```diff
@@ -129,6 +145,11 @@ def execute(self, emulator=None, **overrides):
 
         signal.signal(signal.SIGALRM, alarmHandler)
         signal.alarm(waitTime)
+        try:
```

Comment:
How about we move it to the end of this method and set another alarm for the EOS copy (just logging in case any exception is raised - including Alarm)?

Reply:
Using signal.alarm is dicey. Why not push the timeout logic downward?

Reply:
Are you suggesting to leave the EOS transfer without any timeout (and keep it only for the local logArchive stageout)? 
We still need to have a mechanism to avoid the job hanging on this file transfer, that's why we should have two alarms IMO. We, of course, don't need to wait for 1h * retryNumber, maybe 5 or 10min would suffice.